### PR TITLE
feat/keep tests in separate repo for upgrade to 1.22

### PR DESCRIPTION
**Pull request by mur-me**
## Issue

This PR has all the work from the https://github.com/harmony-one/harmony/pull/4722 PR, except commit that is moving tests to the repo

### Unit Test Coverage

Before:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

After:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **YES|NO**

2. **Describe the migration plan.**. For each flag epoch, describe what changes take place at the flag epoch, the anticipated interactions between upgraded/non-upgraded nodes, and any special operational considerations for the migration.

3. **Describe how the plan was tested.**

4. **How much minimum baking period after the last flag epoch should we allow on Pangaea before promotion onto mainnet?**

5. **What are the planned flag epoch numbers and their ETAs on Pangaea?**

6. **What are the planned flag epoch numbers and their ETAs on mainnet?**

    Note that this must be enough to cover baking period on Pangaea.

7. **What should node operators know about this planned change?**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **YES|NO**

9. **Does the existing `node.sh` continue to work with this change?**

10. **What should node operators know about this change?**

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

### CODE REVIEW

File: cmd/bootnode/main.go
Comment (GheisMohammadi) [inline cmd/bootnode/main.go:111]: I put a comment here to follow up with the rest of 3 cases

File: internal/configs/node/network.go
Comment (GheisMohammadi) [inline internal/configs/node/network.go:83]: and here as well. Let's make sure it is backward compatible with current main-net validators configs.

File: .travis.yml
Comment (Frozen) [inline .travis.yml:32]: Will we return back after PR will be merged?

**Review by Frozen — APPROVED** [review]

File: cmd/bootnode/main.go
Comment (sophoah) [reply cmd/bootnode/main.go:111]: I would recommend to change this to `false` to follow the current mainnet setting and avoid any potential network disruption

File: internal/configs/node/network.go
Comment (sophoah) [reply internal/configs/node/network.go:83]: same comment as above

File: .travis.yml
Comment (sophoah) [reply .travis.yml:32]: @mur-me why are we using your fork and not the harmony github account ?

**Comment by sophoah** [comment]
@Frozen can you explain why we need core/genesis/go19.go, core/genesis/go20.go and why there isn't something for  go 1.22 ?

File: .travis.yml
Comment (mur-me) [reply .travis.yml:32]: 'cause I don't have a write access to the `harmony-test` repo

Comment (sophoah) [reply .travis.yml:32]: ok i see, i created feature/go1.22 on harmony-one/harmony-test repo. can you make the update accordingly here. Thanks

**Comment by Frozen** [comment]
> @Frozen can you explain why we need core/genesis/go19.go, core/genesis/go20.go and why there isn't something for go 1.22 ?

Changed to 1.22

**Comment by Frozen** [comment]
> @Frozen should we also update : [api/service/legacysync/downloader/Proto.Dockerfile](https://github.com/harmony-one/harmony/blob/b3aebb62d29379c4bcb32438051ef40870184b24/api/service/legacysync/downloader/Proto.Dockerfile#L4) ? the dockerfile was added in #4427
> 
> @mur-me there is a few instances of 1.19, please help update https://github.com/search?q=repo%3Aharmony-one%2Fharmony+1.19&type=code

Proto.Dockerfile already updated

**Review by sophoah — APPROVED** [review]

**Review by GheisMohammadi — APPROVED** [review]

**Pull request merged**
